library/waku_thread/config.nim

```diff
@@ -149,6 +165,29 @@ proc parseTopics(jsonNode: JsonNode, conf: var WakuNodeConf) =
   else:
     conf.topics = @["/waku/2/default-waku/proto"]
 
+proc parseRLNRelay(jsonNode: JsonNode, conf: var WakuNodeConf, errorResp: var string): bool =
+  if not jsonNode.contains("rlnRelay"):
+    return true
+
+  let jsonNode = jsonNode["rlnRelay"]
+  if not jsonNode.contains("enabled"):
+    errorResp = "rlnRelay.enabled attribute is required"
+    return false
+
+  conf.rlnRelay = jsonNode["enabled"].getBool()
+  conf.rlnRelayCredPath = jsonNode{"credentialPassword"}.getStr()
+  conf.rlnRelayEthClientAddress = EthRpcUrl.parseCmdArg(jsonNode{"ethClientAddress"}.getStr("http://localhost:8540"))
+  conf.rlnRelayEthContractAddress = jsonNode{"contractAddress"}.getStr()
+  conf.rlnRelayCredPassword = jsonNode{"credentialPassword"}.getStr()
+  conf.rlnRelayUserMessageLimit = uint64(jsonNode{"userMessageLimit"}.getInt(1))
+  conf.rlnEpochSizeSec = uint64(jsonNode{"epochSizeSec"}.getInt(1))
+  conf.rlnRelayCredIndex = some(uint(jsonNode{"membershipIndex"}.getInt()))
+  conf.rlnRelayDynamic = jsonNode{"dynamic"}.getBool()
+  conf.rlnRelayTreePath = jsonNode{"treePath"}.getStr()
+  conf.rlnRelayBandwidthThreshold = jsonNode{"bandwidthThreshold"}.getInt()
```

Comment:
Analogue to my previous comment, I suggest using the same names as in `wakunode` app:
```
$ ./build/wakunode2 --help | grep rln
 --rln-relay-cred-path     The path for peristing rln-relay credential.
 --rln-relay-eth-client-address  HTTP address of an Ethereum testnet client e.g., http://localhost:8540/
 --rln-relay-eth-contract-address  Address of membership contract on an Ethereum testnet.
 --rln-relay-cred-password  Password for encrypting RLN credentials.
 --rln-relay-eth-private-key  Private key for broadcasting transactions.
 --rln-relay-user-message-limit  Set a user message limit for the rln membership registration. Must be a positive
 --rln-relay-epoch-sec     Epoch size in seconds used to rate limit RLN memberships. Default is 1 second.
 --rln-relay               Enable spam protection through rln-relay: true|false [=false].
 --rln-relay-membership-index  the index of the onchain commitment to use.
 --rln-relay-dynamic       Enable  waku-rln-relay with on-chain dynamic group management: true|false
 --rln-relay-id-key        Rln relay identity secret key as a Hex string.
 --rln-relay-id-commitment-key  Rln relay identity commitment key as a Hex string.
 --rln-relay-tree-path     Path to the RLN merkle tree sled db (https://github.com/spacejam/sled).
 --rln-relay-bandwidth-threshold  Message rate in bytes/sec after which verification of proofs should happen [=0].
 --rln-relay-tree-path     Path to the RLN merkle tree sled db (https://github.com/spacejam/sled).
warning: could not open directory 'postgres-data/': Permission denied
```

Reply:
I can use the hyphens to separate the words but adding the `rln` prefix I'm not sure, because it currently looks like this:

```json
{
   "port": 123,
   "host": "0.0.0.0",
   "relay": true,
   "rlnRelay": {
          "enabled": "true",
         "ethClientAddress": "http://localhost:8545",
         "membershipIndex": 2,
         ...
    }
}
```
And seems to me that adding the `rln` prefix would be redundant, as the values are already part of a `rlnRelay` section. A possibility could be to put everything a single level deep (i.e., move all the items from `rlnRelay` into the root json, but seems to me that it's not benefiting from the possibilities of json allowing more logical groups.

Would something like this work?  (i.e. having the `rln-relay` prefix be instead the key that holds all the rln related configs):
```json
```json
{
   "port": 123,
   "host": "0.0.0.0",
   "relay": true,
   "rln-relay": {
          "enabled": "true",
         "eth-client-address": "http://localhost:8545",
         "membership-index": 2,
         "bandwidth-threshold": 1,
         ...
    }
}
```

Reply:
Yes, I think this is the best balanced option (simplicity vs consistency)
![image](https://github.com/waku-org/nwaku/assets/128452529/8e4b0889-250c-4dcd-8e6e-7360768636c4)